contracts/multisig/src/key.rs

```diff
@@ -211,37 +215,40 @@ impl KeyDeserialize for KeyType {
     }
 }
 
-const ECDSA_COMPRESSED_PUBKEY_LEN: usize = 33;
-const ECDSA_UNCOMPRESSED_PUBKEY_LEN: usize = 65;
-
-const ED25519_PUBKEY_LEN: usize = 32;
-const ED25519_SIGNATURE_LEN: usize = 64;
+// It is possible for the same public key to be represented in different formats. For example
+// an ECDSA public key can be represented as compressed or uncompressed. This function
+// validates and normalizes the public key to a single format.
+fn validate_and_normalize_public_key(
```

Comment:
We should just restrict what's allowed to be submitted instead, i.e it should already be in the normalized form and fail if it isn't. And we can control this in `ampd` anyways. This is generally a better cryptographic practice. Also simplifies the effort for any indexers.

Reply:
I am OK with that, too.